standards/core/peer-exchange.md

```diff
@@ -66,26 +66,40 @@ message PeerInfo {
   bytes enr = 1;
 }
 
-message PeerExchangeQuery {
+message PeerExchangeRequest {
   uint64 num_peers = 1;
 }
 
 message PeerExchangeResponse {
   repeated PeerInfo peer_infos = 1;
+  uint32 status_code = 10;
+  optional string status_desc = 11;
 }
 
 message PeerExchangeRPC {
-  PeerExchangeQuery query = 1;
+  PeerExchangeRequest request = 1;
   PeerExchangeResponse response = 2;
 }
 
 ```
 
 The `enr` field contains a Waku ENR as specified in [WAKU2-ENR](./enr.md).
 
-Requesters send a `PeerExchangeQuery` to a peer.
+Requesters send a `PeerExchangeRequest` to a peer.
 Responders SHOULD include a maximum of `num_peers` `PeerInfo` instances into a response.
 Responders send a `PeerExchangeResponse` to requesters containing a list of `PeerInfo` instances, which in turn hold an ENR.
+Responders, MUST include a `status_code` in the response in any case and possible descriptive form of the problem if any.
+
+### Possible status codes
+
+| Result | Code | Note |
+|--------|------|------|
+| SUCCESS  | 200 | Successfull request-respond.  |
+| BAD_REQUEST | 400   | Wrong request payload. It must only contain `request` field.  |
+| BAD_RESPONSE | 401   | Wrong response payload. If success it must contain `response` and `status_code` fields. If failure, only `status_code` is set.  |
+| TOO_MANY_REQUESTS | 429 | DOS protection prevented this request as the current request exceeds the configured request rate. |
+| SERVICE_UNAVAILABLE  | 503 | Request cannot be served, either issue on Responder side or having no suitable peer to issue request.  |
```

Comment:
`r having no suitable peer to issue request` is not valid for a peer-exchange server.
In case of peer-exchange the only scenario where 503 can be returned is when discv5 service is not running and no peers are populated in the cache of server.